### DISCUSSION TITLE
Trying to install OpenGothic on Mac and getting "CMake Error: The source directory does not appear to contain CMakeLists.txt"

### DISCUSSION THREAD

**Discussion by CROATOA88** [Try/OpenGothic#487] [Q&A]
Title. What am I doing wrong? An error appears after "cmake -H. -Bbuild -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo"

**Comment by Try** [comment]
Hi, @CROATOA88 !

When you calling cmake, what is you current working directory?
Also, afaik mac-source-ports project have a prebuilt binary here: https://macsourceports.com/game/gothic2

**Reply by CROATOA88** [reply]
The working directory is /Users/username/Library/Application Support/OpenGothic, it is a directory where I installed GOG Windows version using  Extractor. Mac-source-ports version works, but they stopped updating the build in February.

**Reply by Try** [reply]
> /Users/username/Library/Application Support/OpenGothic

This looks more like data folder.
cmake suppose to be executed in folder, where source code is

**Reply by CROATOA88** [reply]
Choosing the source code folder gives me another set of errors -

CMake Deprecation Warning at lib/edd-dbg/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Error at CMakeLists.txt:87 (add_subdirectory):
  The source directory

    /Users/username/Downloads/OpenGothic-opengothic-v1.0.2207 3/lib/phoenix

  does not contain a CMakeLists.txt file.


CMake Error at CMakeLists.txt:96 (add_subdirectory):
  add_subdirectory given source "lib/Tempest/Engine" which is not an existing
  directory.


CMake Error at CMakeLists.txt:118 (add_subdirectory):
  The source directory

    /Users/username/Downloads/OpenGothic-opengothic-v1.0.2207 3/lib/bullet3

  does not contain a CMakeLists.txt file.


CMake Error at CMakeLists.txt:120 (target_compile_options):
  Cannot specify compile options for target "BulletDynamics" which is not
  built by this project.


CMake Error at CMakeLists.txt:121 (target_compile_options):
  Cannot specify compile options for target "BulletInverseDynamics" which is
  not built by this project.


CMake Error at CMakeLists.txt:122 (target_compile_options):
  Cannot specify compile options for target "BulletSoftBody" which is not
  built by this project.


CMake Error at CMakeLists.txt:123 (target_compile_options):
  Cannot specify compile options for target "Bullet3Common" which is not
  built by this project.


CMake Error at CMakeLists.txt:124 (target_compile_options):
  Cannot specify compile options for target "BulletCollision" which is not
  built by this project.


CMake Error at CMakeLists.txt:125 (target_compile_options):
  Cannot specify compile options for target "LinearMath" which is not built
  by this project.

**Reply by Try** [reply]
You seem to missing all source code for submodules, use `git submodule update --init --recursive` to fix it.
Also note, that we have build instruction in readme with commands

**Reply by CROATOA88** [reply]
That gives "fatal: not a git repository (or any of the parent directories): .git" message.. Well I suppose I'll just have to wait for a prebuilt release

**Reply by Nindaleth** [reply]
Which part of [this guide](https://github.com/Try/OpenGothic#build-on-macos) is the first that fails for you? It seems you tried to do it some other way, which is why you get the errors. Also if you try that, please start in a clean directory to make sure the previous attempts don't cause issues.
